src/libostree/ostree-bootconfig-parser.c

```diff
@@ -168,23 +170,30 @@ ostree_bootconfig_parser_write_at (OstreeBootconfigParser   *self,
       value = g_hash_table_lookup (self->options, key);
       if (value == NULL)
         {
-          g_string_append (buf, line);
+          g_hash_table_insert (lines_overrides, (gpointer)key, (gpointer)line);
+        }
+    }
+
+  for (guint i = 0; i < G_N_ELEMENTS (fields); i++)
+    {
+      const char *key = fields[i];
+      const char *value = NULL;
+
+      value = g_hash_table_lookup (lines_overrides, key);
```

Comment:
> The intention of the current code (I _think_) is to support e.g. preserving comment lines (and

I see. I completely forgot about comment lines in BLS snippets.

> particularly their ordering). Hopefully zipl supports that?

Yes, zipl supports having comment lines in BLS snippets. Or rather, it just ignores them when parsing the BLS files to create its boot options.

> I guess though trying to preserve comment order raises the question - do we _fix_ BLS files that existed before? If we do we risk invalidating comment ordering.

But currently the comment lines are not carried for new deployments, right? At least that's what I see from my tests. They are only preserved when a BLS file with comments is copied to a new /boot/loader.<N> directory after a bootconfig swap.

Does OSTree ever write BLS files with comment? I think that the expected behavior should be to carry these if the user added comments to them by modifying the BLS files without using OSTree?

If the expectation is for OSTree to preserve the comments, then I think that these should be stored in `OstreeBootconfigParser` along with the lines and options, and associate the comments with each option. That way both the options and their associated comments could be sorted when writing new BLS files.

Reply:
> This all said...I also wouldn't be opposed to just dropping support for comments. I doubt anyone relies on them today because, as noted, ostree writes a new entry, so they'll get GC'd anyways.

Indeed. And I guess that modifying the BLS files outside of OSTree is something that should be avoided in general.